File: woodroid/src/main/java/me/gilo/woodroid/repo/ShippingMethodRepository.java
package me.gilo.woodroid.repo;

import retrofit2.Call;
import me.gilo.woodroid.data.api.ShippingMethodAPI;
import me.gilo.woodroid.models.ShippingMethod;

import java.util.List;

public class ShippingMethodRepository extends WooRepository{

    private final ShippingMethodAPI apiService;

    public ShippingMethodRepository(String baseUrl, String consumerKey, String consumerSecret) {
       super( baseUrl, consumerKey, consumerSecret);
       apiService = retrofit.create(ShippingMethodAPI.class);
    }

    public Call<ShippingMethod> shippingMethod(String id) {
        return apiService.view(id);
    }

    public Call<List<ShippingMethod>> shippingMethods() {
        return apiService.list();
    }


}

File: woodroid/src/main/java/me/gilo/woodroid/repo/ShippingMethodRepository.kt
package me.gilo.woodroid.repo

import retrofit2.Call
import me.gilo.woodroid.data.api.ShippingMethodAPI
import me.gilo.woodroid.models.ShippingMethod

class ShippingMethodRepository(baseUrl: String, consumerKey: String, consumerSecret: String) :
    WooRepository(baseUrl, consumerKey, consumerSecret) {

    private val apiService: ShippingMethodAPI

    init {
        apiService = retrofit.create(ShippingMethodAPI::class.java)
    }

    fun shippingMethod(id: String): Call<ShippingMethod> {
        return apiService.view(id)
    }

    fun shippingMethods(): Call<List<ShippingMethod>> {
        return apiService.list()
    }


}
